Issue/22 Improving end-user documentation (#43)
inmanta/vscode-inmanta: README.md

## Features

> ### Syntax coloring

Author: Changed it in https://github.com/inmanta/vscode-inmanta/pull/43/commits/81aa4a44a12aba1a05864a448e0504ecfbef91dd

The extension makes your life easier by coloring syntax keywords or the Inmanta language.

![Syntax coloring screenshot](images/screenshot-syntax-coloring.png)

### Code navigation

> The extension allows you to navigate to the Definition of types used by using the shortcut `Ctrl + Click` when you hover the element.  And will show an overview of the Entity if you don't click.

Author: Did it in https://github.com/inmanta/vscode-inmanta/pull/43/commits/a908809353200281304c8304f69c54b6b892c7aa

![Navigation screenshot](images/screenshot-ctrl-click.png)

### Compilation error reporting
The extension will run a pre-compilation every time you save your file.  Making it easier for you to spot syntax and basic model errors.  When hovering the problematic bit, an explaination message is shown, and more details can be found in the **Output** panel (when selecting *Inmanta Server Language* in the expandable list on the right).

![Error reporting screenshot](images/screenshot-error-reporting.png)

> ### Model export

Reviewer: purely informative: is this a build task? (i.e. if you press f5, does it export?)

Author: We would need at least to add a task to `.vscode/tasks.json` to be able to do that right?

The extension adds a button in the upper right corner of the editing panel of any inmanta model file (ending in `.cf`): **Export to server**.  This buttons allows you to easily export the current open model to an Inmanta server running.  Note that for this to work, you need to have previously configured the server, and added the following informations in the `.inmanta` configuration file at the root of the project:

```
[config]

> fact-expire=1800

Reviewer: ```suggestion
fact-expire=1800
```

not relevant here

Author: Removed it in https://github.com/inmanta/vscode-inmanta/pull/43/commits/a908809353200281304c8304f69c54b6b892c7aa

environment=f2f6ac46-10a2-4e75-897d-b91c51c78df2  # Replace this by the right environment uuid

[compiler_rest_transport]
host=10.0.0.102                                   # Replace this by the address of the host
port=8888                                         # Replace this by the port on which the server is listening

[cmdline_rest_transport]
host=10.0.0.102                                   # Replace this by the address of the host
port=8888                                         # Replace this by the port on which the server is listening
```

## Installation

> To enable ctrl-click for code navigation, a few extra steps are required:

Reviewer: is this still required?

Author: It seems like it is not, indeed.  If not input is given for the `Python Path`, a default one is created and everything will work.

Author: I updated the documentation accordingly here: https://github.com/inmanta/vscode-inmanta/pull/43/commits/8d43418d91ec5cf21b0f6846210a337521fd6668

 - Removed the installation part
 - Added a note below the features requiring the LS, explaining how to enable it.

## Features

### Syntax highlighting
The extension makes your life easier by highlighting syntax keywords or the Inmanta language.

![Syntax coloring screenshot](images/screenshot-syntax-coloring.png)

### Code navigation

> When pressing on `Ctrl` and hovering an element, and overview of its type Entity is shown.  If you click on it (while still having `Ctrl` pressed on) you can navigate to the definition of the entity.

Reviewer 2: I think this reads a bit more easily:
```suggestion
When you hover an element while pressing `Ctrl`, an overview of its Entity type is shown. If you click it (while still having `Ctrl` pressed) you can navigate to the definition of the entity.
```

> :bulb: To work properly, this feature requires the Inmanta Language Server to be running:  

> > In the inmanta extension configuration (`ctrl+, > extensions > inmanta`), set `LS`: to Enabled

Reviewer: shouldn't we enable by default?

Author: I guess it wasn't that useful before when more steps where required for the installation anyway but it could be now?

Author: I don't see any other open issue that would be relevant to include that change if required, so I will wait a bit more to merge this one.

## Extension Settings

You can find the settings of the extension, in the Settings panel, under `Extensions > Inmanta`.

![Settings screenshot](images/screenshot-settings.png)

The extension currently has three settings:

>  - inmanta.compilerVenv: Path to the virtual environment the compiler should use.

Reviewer 2: ```suggestion
 - inmanta.compilerVenv: Absolute path to the virtual environment the compiler should use.
```

 - inmanta.ls.enabled: Whether or not activate the Inmanta Language server (you probably want to).

>  - inmanta.pythonPath: Python path the extension should use.

Reviewer 2: ```suggestion
 - inmanta.pythonPath: Python path the extension should use to run the language server.
```
